test/extended/README.md

```diff
@@ -0,0 +1,145 @@
+# OpenShift extended test suite
+
+This document describes how a developer can write a new extended test for
+OpenShift and the structure of extended tests.
+
+Prerequires
+------------------
+
+In order to execute the extended tests, you have to install
+[Ginkgo](https://github.com/onsi/ginkgo) framework which is used in extended
+tests. You can do it by running following command:
+
+```console
+$ go get github.com/onsi/ginkgo/ginkgo
+```
+
+You also need to have the `openshift` binary in the `PATH` if you want to use
+the shell script helpers to execute the extended tests.
+
+Extended tests structure
+------------------------
+
+Extended tests live under the `./test/extended` directory in the origin repository.
+
+The structure of this directory is following:
+
+* [**`test/extended/util`**](util) provides useful helpers and utilities to use in your extended test. It provides a easy-to-use interface to OpenShift CLI and also
+access to the Kubernetes [E2E framework](https://github.com/openshift/origin/tree/master/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e) helpers. It also contains OpenShift helpers are shared across multiple test cases, to make the test cases more DRY.
+* [**`test/extended/fixtures`**](fixtures) contains the JSON and YAML fixtures are meant to be used by the extended tests.
+* [**`test/extended/[images,builds,...]`**](builds) each of this Go packages contains extended tests that are related together. For example, the `images` directory should contain test cases that are exercising usage of various Docker images in OpenShift.
+* [**`hack/test-extended/[group]/run.sh`**](../../hack/test-extended) contains a shell script that helps you to execute extended tests for a certain group
```

Comment:
s/contains a shell script that helps you to execute extended tests for a certain group/ is the shell script that sets up any needed dependencies and then launches the extended tests whose top level ginkgo spec's Describe call starts with "[group]:"